Core/OxidEE_Events.php

```diff
@@ -113,18 +117,38 @@ private static function _extendPaymentMethodTable()
             " ADD COLUMN `WDOXIDEE_APIURL` varchar(128) default '' NOT NULL";
         self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_APIURL', $sQueryAddApiUrl);
 
-        $sQueryAddMaid = "ALTER TABLE " . self::PAYMENT_TABLE .
-            " ADD COLUMN `WDOXIDEE_MAID` varchar(128) default '' NOT NULL";
-        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_MAID', $sQueryAddMaid);
-
         $sQueryAddIsWirecard = "ALTER TABLE " . self::PAYMENT_TABLE .
             " ADD COLUMN `WDOXIDEE_ISWIRECARD` tinyint(1) default 0 NOT NULL";
         self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_ISWIRECARD', $sQueryAddIsWirecard);
 
+        $sQueryAddMaid = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_MAID` varchar(128) default '' NOT NULL";
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_MAID', $sQueryAddMaid);
+
         $sQueryAddSecret = "ALTER TABLE " . self::PAYMENT_TABLE .
             " ADD COLUMN `WDOXIDEE_SECRET` varchar(128) default '' NOT NULL";
         self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_SECRET', $sQueryAddSecret);
 
+        $sQueryAddThreeDMaid = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_THREE_D_MAID` varchar(128) default '' NOT NULL";
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_THREE_D_MAID', $sQueryAddThreeDMaid);
+
+        $sQueryThreeDSecret = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_THREE_D_SECRET` varchar(128) default '' NOT NULL";
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_THREE_D_SECRET', $sQueryThreeDSecret);
+
+        $sQueryAddMaxLimit = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_NON_THREE_D_MAX_LIMIT` varchar(128) default '' NOT NULL";
```

Comment:
We should use an **integer** representation instead as the 3d limit inputs only allow numbers as far as I can see.

Reply:
changed to double